### ISSUE TITLE
PHP highlighting breaks on here-docs

### ISSUE THREAD

**Issue by gnprice** [isagalaev/highlight.js#888]
On PHP code that contains a here-doc, the "string" highlight for the here-doc continues on long after the here-doc has actually been closed -- it seems to miss the end of the here-doc.

For example if this snippet is highlighted as PHP:
```
<?php
    print(<<<EOTEXT
      this is a heredoc ...
EOTEXT
        );
    print("... but this line is code.");
?>
```
then everything from `<<<EOTEXT` to the end goes in a span of class `hljs-string`.

See repro at https://jsfiddle.net/5pdjee5v/ .

(/cc @christoffer, and thanks @jboning for spotting the issue!)

**Comment by isagalaev** [comment]
As currently defined, it wants a semicolon at the end of "EOTEXT" here. Is this wrong?

**Comment by jhurwitz** [comment]
From what I can tell, heredocs should end with a semicolon unless used as a function argument (as in the example given above). See also example #3 here: http://php.net/manual/en/language.types.string.php#language.types.string.syntax.heredoc

**Comment by jboning** [comment]
Even in statements, it seems to work fine without a trailing semicolon on the same line:
```
php > $a = <<<EOTEXT
<<< > foo
<<< > EOTEXT
php > ;
php > print $a;
foo
```

**Comment by Sannis** [comment]
Yeah, many rare use-cases are not highlighted properly in many language and we always try to improve them on bug reports if it is not overkill.

**Comment by isagalaev** [comment]
Fixed (sort of) in a9e6bd6 (see commit message).

**Comment by Sannis** [comment]
Thanks!
